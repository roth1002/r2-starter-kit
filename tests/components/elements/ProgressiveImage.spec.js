import React from 'react';
import { mount } from 'enzyme';
import Component from 'components/elements/ProgressiveImage';

describe('<ProgressiveImage />', () => {
  const initProps = {
    url: 'https://placehold.it/800x600',
    placeholder: 'https://placehold.it/40x30',
  };

  let wrapper;
  beforeAll(() => {
    wrapper = mount(<Component {...initProps} />);
  });

  it('should render initial component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should replace fuzzy tiny image with sharp large one after image onload', () => {
    wrapper.find('img').at(1).simulate('load');
    expect(wrapper).toMatchSnapshot();
  });
});
