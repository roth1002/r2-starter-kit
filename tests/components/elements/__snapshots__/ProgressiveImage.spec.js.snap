// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressiveImage /> should render initial component 1`] = `
<ProgressiveImage
  placeholder="https://placehold.it/40x30"
  url="https://placehold.it/800x600"
>
  <div>
    <img
      className="placeholder"
      style={
        Object {
          "backgroundImage": "url('https://placehold.it/40x30')",
        }
      }
    />
    <img
      onLoad={[Function]}
      src="https://placehold.it/800x600"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</ProgressiveImage>
`;

exports[`<ProgressiveImage /> should replace fuzzy tiny image with sharp large one after image onload 1`] = `
<ProgressiveImage
  placeholder="https://placehold.it/40x30"
  url="https://placehold.it/800x600"
>
  <div>
    <img
      className="placeholder sharp"
      style={
        Object {
          "backgroundImage": "url('https://placehold.it/800x600')",
        }
      }
    />
    <img
      onLoad={[Function]}
      src="https://placehold.it/800x600"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </div>
</ProgressiveImage>
`;
